### PR TITLE
scrub: use a generic interface for scheduling timer based events

### PR DESCRIPTION
**Pull request by athanatos**
<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by athanatos** [comment]
Ok, this version passes osd-scrub-test.sh.  I also added a couple of commits addressing minor things with the test itself.

File: src/osd/scrubber/pg_scrubber.cc [hunk 1]
```diff
@@ -819,89 +851,6 @@ bool PgScrubber::range_intersects_scrub(const hobject_t& start,
...
-/**
- *  if we are required to sleep:
- *	arrange a callback sometimes later.
- *	be sure to be able to identify a stale callback.
- *  Otherwise: perform a requeue (i.e. - rescheduling thru the OSD queue)
- *    anyway.
- */
-void PgScrubber::add_delayed_scheduling()
-{
-  m_end = m_start;  // not blocking any range now
```
[the remark this answers is not in the document]
Comment (ronen-fr) [reply src/osd/scrubber/pg_scrubber.cc:852]: Ah. I see the commit comment.

**Review by ronen-fr — APPROVED** [review]
I like this change...

**Comment by ljflores** [comment]
jenkins test api

**Comment by ljflores** [comment]
http://pulpito.front.sepia.ceph.com/?branch=wip-yuri7-testing-2023-02-27-1105

Failures, unrelated:
    1. https://tracker.ceph.com/issues/58585
    2. https://tracker.ceph.com/issues/58475
    3. https://tracker.ceph.com/issues/57754
    4. https://tracker.ceph.com/issues/50786
    5. https://tracker.ceph.com/issues/49287

Details:
    1. rook: failed to pull kubelet image - Ceph - Orchestrator
    2. test_dashboard_e2e.sh: Conflicting peer dependency: postcss@8.4.21 - Ceph - Mgr - Dashboard
    3. test_envlibrados_for_rocksdb.sh: update-alternatives: error: alternative path /usr/bin/gcc-11 doesn't exist - Ceph - RADOS
    4. UnicodeDecodeError: 'utf8' codec can't decode byte - Ceph - RADOS
    5. SELinux Denials during cephadm/workunits/test_cephadm - Ceph - Orchestrator

**Pull request merged**